infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/environment/util/EntryPointParser.java

```diff
@@ -52,6 +53,21 @@ public EntryPoint parse(Map<String, String> machineAttributes) throws Infrastruc
     return new EntryPoint(commandList, argList);
   }
 
+  /**
+   * Serializes an entry (that might have been produced from {@link #parse(Map)}) back to a string
+   * representation.
+   *
+   * @param entry the command or args entry
+   * @return a serialized representation of the entry
+   */
+  public String serializeEntry(List<String> entry) {
+    try {
+      return mapper.writer().writeValueAsString(entry);
+    } catch (JsonProcessingException e) {
+      throw new IllegalArgumentException(e);
```

Comment:
When would this exception be thrown / can it be caused by user input? If so I would prefer a checked exception here (even if it's a `DevfileException` or something) since unchecked means more to think about.

Reply:
:+1:

Reply:
This should not be caused by any user input. Actually we should throw something like `IllegalStateException` because this really should not happen.